Running: /home/sarnold/bin/audit-code.sh -c python ./
Scanning './'
== subprocess_spawned() ==
./docs/conf.py:23:os.system('%s setup.py test -q' % sys.executable)

== memory_management() ==
(unsupported type 'python')

== files() ==
./setup.py:25:README = open(os.path.join(here, 'README.txt')).read()
./setup.py:26:CHANGES = open(os.path.join(here, 'CHANGES.txt')).read()
./ez_setup.py:146:            src = urllib2.urlopen(url)
./ez_setup.py:150:            dst = open(saveto,"wb"); dst.write(data)
./ez_setup.py:240:        f = open(name,'rb')
./ez_setup.py:250:    f = open(srcfile, 'rb'); src = f.read(); f.close()
./ez_setup.py:258:    f = open(srcfile,'w')
./repoze/who/config.py:138:    parser.parse(open(config_file))
./repoze/who/config.py:145:            log_stream = open(log_file, 'wb')
./repoze/who/tests/test_config.py:365:        config = open(path, 'w')
./repoze/who/plugins/tests/test_form.py:142:        formbody = open(form).read()
./repoze/who/plugins/tests/test_form.py:196:        formbody = open(form).read()
./repoze/who/plugins/tests/test_authtkt.py:341:        secret = open(path, 'w')
./repoze/who/plugins/htpasswd.py:28:                f = open(self.filename, 'r')
./repoze/who/plugins/auth_tkt.py:206:        secret = open(secretfile).read().strip()
./repoze/who/plugins/form.py:219:        form = open(form).read()

== logging() ==
./ez_setup.py:129:                log.warn("""
./ez_setup.py:145:            log.warn("Downloading %s", url)
./repoze/who/plugins/tests/test_htpasswd.py:97:            def warn(self, msg):
./repoze/who/plugins/htpasswd.py:30:                environ['repoze.who.logger'].warn('could not open htpasswd '
./repoze/who/middleware.py:164:                logger and logger.debug(
./repoze/who/middleware.py:168:                logger and logger.debug(
./repoze/who/middleware.py:171:        logger and logger.debug('identities found: %s' % (results,))
./repoze/who/middleware.py:203:                    logger and logger.debug(
./repoze/who/middleware.py:213:                    logger and logger.debug(
./repoze/who/middleware.py:219:        logger and logger.debug('identities authenticated: %s' % (results,))

== environment() ==
./repoze/who/tests/test_middleware.py:891:    def test_it_no_WHO_LOG_in_environ(self):
./repoze/who/tests/test_middleware.py:904:    def test_it_w_WHO_LOG_in_environ(self):
./repoze/who/plugins/tests/test_form.py:443:    def test_challenge_came_from_in_environ(self):
./repoze/who/plugins/tests/fixtures/testapp.py:1:def tack_environ(environ, msg):
./repoze/who/plugins/tests/fixtures/testapp.py:8:    msg = tack_environ(environ, msg)
./repoze/who/plugins/tests/fixtures/testapp.py:17:    msg = tack_environ(environ, msg)

== privileged() ==

== networking() ==
./setup.py:25:README = open(os.path.join(here, 'README.txt')).read()
./setup.py:26:CHANGES = open(os.path.join(here, 'CHANGES.txt')).read()
./ez_setup.py:149:            data = _validate_md5(egg_name, src.read())
./ez_setup.py:150:            dst = open(saveto,"wb"); dst.write(data)
./ez_setup.py:241:        md5_data[base] = md5(f.read()).hexdigest()
./ez_setup.py:250:    f = open(srcfile, 'rb'); src = f.read(); f.close()
./ez_setup.py:259:    f.write(src)
./repoze/who/tests/test_middleware.py:771:        def write(data):
./repoze/who/tests/test_config.py:366:        config.write(text)
./repoze/who/plugins/tests/test_form.py:142:        formbody = open(form).read()
./repoze/who/plugins/tests/test_form.py:196:        formbody = open(form).read()
./repoze/who/plugins/tests/test_authtkt.py:342:        secret.write('s33kr1t\n')
./repoze/who/plugins/auth_tkt.py:206:        secret = open(secretfile).read().strip()
./repoze/who/plugins/sql.py:30:        return psycopg2.connect(kw['repoze.who.dsn']) #pragma NO COVERAGE
./repoze/who/plugins/form.py:219:        form = open(form).read()
./repoze/who/middleware.py:347:                write(value)
./repoze/who/middleware.py:394:        io.write('%s:%s\n' % (name, password))
./repoze/who/plugins/tests/test_form.py:424:        import cgi
./repoze/who/plugins/tests/test_form.py:456:        import cgi
./repoze/who/plugins/tests/test_form.py:483:        import cgi
./repoze/who/plugins/tests/test_form.py:514:        import cgi
./repoze/who/plugins/form.py:3:import cgi

== http:// ==
./setup.py:7:# http://www.repoze.org/LICENSE.txt.  A copy of the license should accompany
./setup.py:44:      url="http://www.repoze.org",
./setup.py:45:      license="BSD-derived (http://www.repoze.org/LICENSE.txt)",
./ez_setup.py:18:DEFAULT_URL     = "http://pypi.python.org/packages/%s/s/setuptools/" % sys.version[:3]
./repoze/who/tests/test_middleware.py:669:            replace_app = HTTPFound('http://example.com/redirect')
./repoze/who/tests/test_middleware.py:687:                         ('location', 'http://example.com/redirect'))
./repoze/who/tests/test_config.py:217:        # see http://bugs.repoze.org/issue92
./repoze/who/plugins/tests/test_form.py:90:        self.assertEqual(app.location(), 'http://localhost:8080/protected')
./repoze/who/plugins/tests/test_form.py:102:        self.assertEqual(app.location(), 'http://localhost:8080/protected')
./repoze/who/plugins/tests/test_form.py:169:                               [('Location', 'http://foo/bar')],
./repoze/who/plugins/tests/test_form.py:175:        self.assertEqual(headers[0], ('Location', 'http://foo/bar'))
./repoze/who/plugins/tests/test_form.py:227:    def _makeOne(self, login_form_url='http://example.com/login.html',
./repoze/who/plugins/tests/test_form.py:287:                                        came_from='http://example.com')
./repoze/who/plugins/tests/test_form.py:294:        self.assertEqual(value, 'http://example.com')
./repoze/who/plugins/tests/test_form.py:302:                                    came_from='http://example.com',
./repoze/who/plugins/tests/test_form.py:311:        self.assertEqual(value, 'http://example.com')
./repoze/who/plugins/tests/test_form.py:345:        environ['HTTP_REFERER'] = 'http://foo.bar'
./repoze/who/plugins/tests/test_form.py:352:        self.assertEqual(value, 'http://foo.bar')
./repoze/who/plugins/tests/test_form.py:360:                                        came_from='http://example.com')
./repoze/who/plugins/tests/test_form.py:366:        self.assertEqual(environ['came_from'], 'http://example.com')
./repoze/who/plugins/tests/test_form.py:385:        environ['HTTP_REFERER'] = 'http://example.com/referer'
./repoze/who/plugins/tests/test_form.py:391:        self.assertEqual(environ['came_from'], 'http://example.com/referer')
./repoze/who/plugins/tests/test_form.py:434:        self.assertEqual(came_from_value, 'http://www.example.com/?default=1')
./repoze/who/plugins/tests/test_form.py:446:        environ['came_from'] = 'http://example.com/came_from'
./repoze/who/plugins/tests/test_form.py:466:        self.assertEqual(came_from_value, 'http://example.com/came_from')
./repoze/who/plugins/tests/test_form.py:471:        environ['came_from'] = 'http://example.com/came_from'
./repoze/who/plugins/tests/test_form.py:495:        self.assertEqual(came_from_value, 'http://example.com/came_from')
./repoze/who/plugins/tests/test_form.py:502:        environ['came_from'] = 'http://example.com/came_from'
./repoze/who/plugins/tests/test_form.py:526:        self.assertEqual(came_from_value, 'http://example.com/came_from')

== encryption ==

== sql() ==
./repoze/who/plugins/tests/test_sql.py:143:    def test_noquery(self):
./repoze/who/plugins/tests/test_sql.py:184:    def test_no_query(self):
./repoze/who/plugins/tests/test_sql.py:231:    def execute(self, query, *arg):
./repoze/who/plugins/sql.py:51:        curs.execute(self.query, identity)
./repoze/who/plugins/sql.py:76:        curs.execute(self.query, identity)

== tmp() ==

== priv_cmds (sudo, gksu, pkexec) ==

== envvars() ==

== webkit() ==

== comments (XXX, FIXME, TODO) ==
./repoze/who/tests/test_middleware.py:719:    # XXX need more call tests:
./repoze/who/plugins/form.py:96:            # XXX we need to replace wsgi.input because we've read it

== unsafe input mechanisms ==
(unsupported type 'python')

